src/js/base/InputPassword.tsx

```diff
@@ -1,15 +1,20 @@
+import { InputContainer } from "@base/InputContainer";
+import { InputIcon } from "@base/InputIcon";
 import React, { useState } from "react";
 import { Input } from "./Input";
-import { InputContainer } from "./InputContainer";
-import { InputIcon } from "./InputIcon";
 
-export const InputPassword = props => {
+type InputPasswordProps = {
+    id: string;
+    name: string;
+};
+
+export const InputPassword = React.forwardRef<HTMLInputElement, InputPasswordProps>((props, ref) => {
     const [show, setShow] = useState(false);
 
     return (
-        <InputContainer align="right">
-            <Input name="password" aria-label="password" type={show ? "text" : "password"} {...props} />
+        <InputContainer>
+            <Input as="input" {...props} ref={ref} type={show ? "" : "password"} />
```

Comment:
Why the `as`? Shouldn't the underlying type of `Input` be `input`?

Reply:
Not sure why this was happening, but without the `as="input"`, the input wouldn't register the text from react-hook-form.

Reply:
Hmm interesting, is `Input` maybe not forwarding the ref to the html `input`? I think this is worth fixing, having to overwrite with `as` seems kinda brutal